woocommerce/woocommerce-ios
# [Order Creation] Design tweaks and landscape phone support for totals panel on Order Form

Closes: #11467
Merge after: #11498 


## Description


As part of improving the Order Form on tablets, we are improving the use of vertical space in the Order Form.

To ensure clarity on the larger screen, we want to ensure that the total and CTA button are always visible, regardless of how many products are added to the order. On #11498, we moved the totals and button to a new expandable sheet to improve that.

This PR addresses the known issues mentioned in #11498:

1. Show Order Total in the collapsed state
2. Show dividers between the button, order total, and other totals, but not at the top of the view
3. Style the Add shipping/Add Coupon buttons as foreground rows
4. Remove the navigation bar Done button in edit mode
5. Landscape safe area inset support for the panel

As part of the landscape support we had to add a `scrollVerticallyIfNeeded` modifier, which doesn't play perfectly with the swipe gesture. This could definitely be improved, but the scroll view "wins", so it's accessible and good enough for now (IMO).

Improving when the scroll view is applied would be a good follow up, and could tie in to making that modifier better.

## Testing instructions


On iPhone:

1. Launch the app
2. Navigate to the `Orders` tab
3. Tap `+`
4. Tap the chevron; observe that the sheet expands to show the Order Total
5. Add a product; observe that the sheet expands to fit its new content
6. Turn the phone to landscape; observe that the sheet shows as expected, some content is still visible, and if needed, the panel content scrolls.
7. Turn back to portrait
8. Scroll the Order Form behind the sheet; observe that you can access all the controls


General: Observe that the screen matches the designs { 6kkCwI9VOEYcidp6UT5bln-fi-192%3A27224 } – there's more to do with some other tweaks that Joe has mentioned, but we've got the original ones sorted here!


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/2472348/babdb800-49c0-4749-8473-4e33d91ef518



---
